python/BioSimSpace/Sandpit/Exscientia/_SireWrappers/_system.py

```diff
@@ -1702,6 +1758,47 @@ def _set_water_topology(self, format, property_map={}):
                 self._sire_object = _SireIO.setGromacsWater(
                         self._sire_object, water_model, property_map)
 
+    def _set_atom_index_tally(self):
```

Comment:
Are these always regenerated on system modification?

Reply:
It's called internally by whatever functions need the tally information, i.e. it's always called as the first thing within `getAtom`, or when getting the index of an atom with `getIndex`. It will be inconsistent if the system is updated, but is always consistent at the point of use.